eth/handler_eth.go

```diff
@@ -34,7 +34,24 @@ import (
 type ethHandler handler
 
 func (h *ethHandler) Chain() *core.BlockChain { return h.chain }
-func (h *ethHandler) TxPool() eth.TxPool      { return h.txpool }
+
+// NilPool satisfies the TxPool interface but does not return any tx in the
+// pool. It is used to disable transaction gossip.
+type NilPool struct {
+}
+
+// NilPool Get always returns nil
+func (n NilPool) Get(hash common.Hash) *types.Transaction {
+	return nil
+}
+
+func (h *ethHandler) TxPool() eth.TxPool {
```

Comment:
We should also make `(h *ethHandler) AcceptTxs() bool` return false if `disableTxGossip` is true, to not accept incoming txs.

Reply:
Done.